### DISCUSSION TITLE
Sidebar with draggable node templates

### DISCUSSION THREAD

**Discussion by teith** [xyflow/xyflow#2928] [Q&A]
Hello everyone! I've encountered the need to create a Sidebar with node templates, which, when moved to the React Flow space, would create new nodes. Is there a similar implementation in React Flow? How can I best implement this?

Thank you!

**Comment by bcakmakoglu** [comment]
Theres an example for that: https://reactflow.dev/docs/examples/interaction/drag-and-drop/

**Reply by teith** [reply]
But this way doesn't work with mobile version, do it?

**Reply by moklick** [reply]
we try to keep the examples as simple as possible. If you need mobile support, you can use a drag n drop library like https://github.com/react-grid-layout/react-draggable or implement it with onTouchStart, onTouchMove and onTouchEnd.
